Translate to english (#15)
geut/dat-workshop: problems/en/05/README.md

# 5 - From a distributed log to a distributed db 

Congratulations :tada::fireworks:!!

If you are here it means you have become a hypercore padawan. It's time to increase your P2P knowledge and follow the road to be a great Jedi.

We just saw that hypercore gives us a distributed log that we can work with. :cool:

> But there are situations when that is not enough. Luckily to us, there are more libs written on top of hypercore and one of those is here to help, meet [hyperdb](/hyperdb) :star:

Reviewer: ```suggestion
But there are situations when that is not enough. Lucky for us, there are more libs written on top of hypercore and one of those is here to help, meet [hyperdb](/hyperdb) :star:
```

Reviewer 2: Should be "Luckily for"

## Hyperdb

> Hyperdb offers a key-value distributed over a set of hypercores database. :cool:

Reviewer: ```suggestion
Hyperdb offers a key-value distributed over a set of hypercores databases. :cool:
```

> One of the most important features is that it give us the ability to have `multiple-writers`. Which takes Dat to the next level, where multiple users, previously authorized now could be able to modify a Dat resource. Ie, **we have multiple owners**.

Reviewer: ```suggestion
One of the most important features is the ability to have `multiple-writers`. This takes Dat to the next level, where multiple **previously authorized** users are now able to modify a Dat resource. I.e., **we have multiple owners**.
```

## A word about what we are building 

> On this workshop we are learning how to use Dat to create a _P2P web app_. More specifically, our goal here is to build together a P2P chat and the codename will be _Olaf_ :smiley_cat:.

Reviewer: ```suggestion
In this workshop, we are learning how to use Dat to create a _P2P web app_. More specifically, our goal here is to build a P2P chat and its codename will be _Olaf_ :smiley_cat:.
```

> So, to build our chat app we are going to use all the things that we have learned. In this module, we are about to introduce a fundamental part, the core of our chat, and it has a codename too: _Saga_ :heart_eyes_cat:.

Reviewer: ```suggestion
So, to build our chat app we are going to use all the things that we have learned. In this module we are about to introduce a fundamental part, the core of our chat, and it has a codename too: _Saga_ :heart_eyes_cat:.
```

## Saga

![saga](images/saga.jpg)

We are going to start using hyperdb to build _saga_.

Ok, so first let's define a simple API that permits multiple users to write messages over a distributed hyperdb.

Let's focus on a list of possible requirements our chat can have:

1. We should be able to write messages.
1. We should be able to read messages in _real-time_.
1. We should be able to detect when an user connects and disconnects from a channel (_room_).

Cool, let's start writing this new API, **saga**. :heart_eyes_cat:

## Exercise 

1. Add a constructor to the class `Saga` that receives a `storage`, a `key` and a `username`.
2. With the `storage` and the `key`, create an instance of hyperdb using **json** as the valueEncoding.

> 3. `Saga` instance should allows us to access to:

Reviewer: ```suggestion
3. `Saga` instance should allow us access to:
```
